### PR TITLE
feat(assistant): Initial implementation of assistants api

### PR DESCRIPTION
**Pull request by christ66**
**Description**

This PR implements one of the apis described in: #1273

**Notes for Reviewers**
Still a WIP, and needs some tests.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: api/openai/assistant.go [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+	Name         string            `json:"name,omitempty"`         // The name of the assistant.
+	Description  string            `json:"description,omitempty"`  // The description of the assistant.
+	Instructions string            `json:"instructions,omitempty"` // The system instructions that the assistant uses.
+	Tools        []Tool            `json:"tools,omitempty"`        // A list of tools enabled on the assistant.
+	FileIDs      []string          `json:"file_ids,omitempty"`     // A list of file IDs attached to this assistant.
+	Metadata     map[string]string `json:"metadata,omitempty"`     // Set of key-value pairs attached to the assistant.
+}
+
+var (
+	assistants = []Assistant{} // better to return empty array instead of "null"
```
Comment (mudler) [inline api/openai/assistant.go:55]: shall we save this state to a file as well? probably we could start thinking about a db directory where to store the json data at this point (same for the file endpoint)

**Comment by mudler** [comment]
@christ66 do you need any help? If you want we can merge this as-is and tackle the other points separately

**Comment by christ66** [comment]
I should have some tests by tomorrow for this. I removed draft so at least it can be merged.

**Review by mudler — APPROVED** [review]
Nice job @christ66 !

**Pull request merged**